blockchain/tx_pool.go

```diff
@@ -924,10 +933,114 @@ func (pool *TxPool) HandleTxMsg(txs types.Transactions) {
 	if pool.config.DenyRemoteTx {
 		return
 	}
+
+	// Filter spam txs based on to-address of failed txs
+	spamThrottler := GetSpamThrottler()
+	if spamThrottler != nil {
+		pool.mu.RLock()
+		poolSize := uint64(len(pool.all))
+		pool.mu.RUnlock()
+
+		// Activate spam throttler when pool has enough txs
+		if poolSize > uint64(spamThrottler.config.ActivateTxPoolSize) {
+			allowTxs, throttleTxs := spamThrottler.classifyTxs(txs)
+
+			for _, tx := range throttleTxs {
+				select {
+				case spamThrottler.throttleCh <- tx:
+				default:
+					logger.Trace("drop a tx when throttleTxs channel is full", "txHash", tx.Hash())
```

Comment:
If we add a counter metric for dropped tx, It can be helpful.

Reply:
added